sympy/combinatorics/fp_groups.py

```diff
@@ -494,6 +494,71 @@ def elements(self):
         P, T = self._to_perm_group()
         return T.invert(P._elements)
 
+def subgroup_quotient(G, H, fp_group=None):
+    '''
+    Compute the quotient group G/H
+    where, H is the subgroup of G.
+    '''
+    # If no parent group is specified,
+    # G is set to the parent `FpGroup`
+    if not fp_group:
+        fp_group = G
+    free_group = fp_group.free_group
+
+    if not isinstance(fp_group, FpGroup):
+        raise ValueError("The parent group must be an instance"
+                                "of FpGroup")
+
+    if not (G.free_group == free_group
+                or H.free_group == free_group):
```

Comment:
If you pass `G` and `H` as lists, then they have no attribute `free_group`, and if you pass them as `FpGroup`s, then in general, their presentations won't be on `free_group`. You could theoretically make `G` and `H` instances of `FpSubgroup`, in which case this check would work. However, it still wouldn't be write to say `q_group = FpGroup(free_group, q_relators)` at the end - it works when `fp_group == G`, but not otherwise.

Reply:
> You could theoretically make G and H instances of FpSubgroup 

We could probably make the user define the free group. Coz' this problem might only be during the polycyclic series computation in which we pass the lists, in all other cases the user would mostly pass them as the instances of the `FpGroup`, right? This could be avoided if we add an argument for the `free_group`. Or maybe we could add it as a default argument and check if `free_group` is defined when both `G` and `H` are lists (or maybe, only `G`), if it isn't we could raise a `ValueError`.